src/pal/src/synchobj/mutex.cpp

```diff
@@ -1317,15 +1315,16 @@ void NamedMutexProcessData::Close(bool isAbruptShutdown, bool releaseSharedData)
     }
 
     // Delete the lock file, and the session directory if it's not empty
-    char path[SHARED_MEMORY_MAX_FILE_PATH_CHAR_COUNT + 1];
-    SIZE_T sessionDirectoryPathCharCount = SharedMemoryHelpers::CopyString(path, 0, SHARED_MEMORY_LOCK_FILES_DIRECTORY_PATH);
-    path[sessionDirectoryPathCharCount++] = '/';
+    PathCharString path;
+    SharedMemoryHelpers::CopyPath(path, SHARED_MEMORY_LOCK_FILES_DIRECTORY_NAME);
     SharedMemoryId *id = m_processDataHeader->GetId();
-    sessionDirectoryPathCharCount = id->AppendSessionDirectoryName(path, sessionDirectoryPathCharCount);
-    path[sessionDirectoryPathCharCount++] = '/';
-    SharedMemoryHelpers::CopyString(path, sessionDirectoryPathCharCount, id->GetName(), id->GetNameCharCount());
+    VerifyStringOperation(path.Append('/')
+        && id->AppendSessionDirectoryName(path)
+        && path.Append('/'));
+    SIZE_T sessionDirectoryPathCharCount = path.GetCount();
+    VerifyStringOperation(path.Append(id->GetName(), id->GetNameCharCount()));
     unlink(path);
-    path[sessionDirectoryPathCharCount] = '\0';
+    path.CloseBuffer(sessionDirectoryPathCharCount);
     rmdir(path);
 #endif // !NAMED_MUTEX_USE_PTHREAD_MUTEX
```

Comment:
@jkotas so this entire part didn't throw exceptions before but now it can - in the case of out of memory. This is in the Close part of mutex which might be called during dispose/cleanup. 

Is this something acceptable ? Or if out of memory should be just return without exception ? Any logging which can be done ?

Reply:
The typical solutions for errors during cleanup is to ignore them and keep trying to clean up as much as possible. Notice that the existing code is ignoring errors from `unlink` or `rmdir` as well.

Reply:
I understand. But in this case, if we have string errors, then I can't trust the string, so I would need to abort the cleanup completely. Do we do any logging in this case ?